### ISSUE TITLE
no-useless-constructor: false positive on constructors declaring TypeScript parameter properties (typescript-eslint exempts them)

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25806]
oxlint `1.78.0`, rule `no-useless-constructor: error`, TS file:

```ts
class Base {
	protected readonly deps: unknown[]
	constructor(...deps: unknown[]) {
		this.deps = deps
	}
}

export class Child extends Base {
	constructor(private readonly dep: object) {
		super(dep)
	}

	use(): object {
		return this.dep
	}
}
```

oxlint reports on `Child`'s constructor:

> `error eslint(no-useless-constructor): Redundant super call in constructor — Remove this constructor or add code to it.`

The constructor is not useless: `private readonly dep` is a **parameter property** — it declares and assigns `this.dep`. Following the fix deletes the field, breaks `use()`, and no longer typechecks under `strict`.

typescript-eslint's extension rule (`@typescript-eslint/no-useless-constructor`, checked on 8.67.0 / ESLint 9.39.5) exempts constructors whose parameters carry accessibility modifiers or `readonly` for exactly this reason, while still flagging genuinely-useless ones (it agrees with oxlint on a plain forwarding constructor without modifiers).

Expected: skip the report when any constructor parameter is a parameter property (has `public`/`protected`/`private`/`readonly` or a decorator), matching typescript-eslint.
